### PR TITLE
Add type_safe/0.2.1

### PR DESCRIPTION
**Pull request by Manu343726**
Specify library name and version:  **type_safe/0.2.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

#1467 must be merged first.

### CODE REVIEW

File: recipes/type_safe/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    settings = 'compiler'
+
+    no_copy_source = True
+    _source_subfolder = 'source_subfolder'
+
+    requires = 'debug_assert/1.3.3'
+
+    @property
+    def repo_folder(self):
```
Comment (uilianries) [inline recipes/type_safe/all/conanfile.py:20]:
```suggestion
    def _repo_folder(self):
```

File: recipes/type_safe/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        extracted_dir = self.name +  "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, '11')
+
+    def package(self):
+        self.copy("*LICENSE", dst="licenses", keep_path=False)
+        self.copy("*", src=os.path.join(self.repo_folder, 'include'), dst='include/')
```
Comment (uilianries) [inline recipes/type_safe/all/conanfile.py:34]:
```suggestion
        self.copy("*", src=os.path.join(self._repo_folder, 'include'), dst='include/')
```

File: recipes/type_safe/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+project(test_package)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
+set(CMAKE_CXX_STANDARD 11)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/type_safe/all/test_package/CMakeLists.txt:11]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

File: recipes/type_safe/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
+set(CMAKE_CXX_STANDARD 11)
+
```
Comment (uilianries) [inline recipes/type_safe/all/test_package/CMakeLists.txt:6]:
```suggestion
```

**Comment by Manu343726** [comment]
@uilianries I think I have resolved all your comments (I'm not 100% sure...), and since #1467 was already merged I think we could try this again. @danimtb is there any public way to force the rebuild of a package? I can `git push --force`, but that seems hacky.

**Comment by danimtb** [comment]
Thanks for asking @Manu343726. Ther is no way to manually retrigger the CI but I will gladly do it for you. Jenkins doesn't like close-opening nor force pushing very much, so please avoid it when possible 😄

**Comment by uilianries** [comment]
Thanks a lot @Manu343726 ! Maybe you can push an empty commit `git commit --allow-empty`

**Pull request merged**